src/test/kotlin/com/github/avrokotlin/avro4k/endecode/EnDecoder.kt

```diff
@@ -0,0 +1,125 @@
+import kotlinx.serialization.KSerializer
+import kotlinx.serialization.SerializationStrategy
+import kotlinx.serialization.serializer
+import org.apache.avro.Schema
+import org.apache.avro.generic.GenericDatumReader
+import org.apache.avro.generic.GenericDatumWriter
+import org.apache.avro.generic.GenericRecord
+import org.apache.avro.io.Decoder
+import org.apache.avro.io.DecoderFactory
+import org.apache.avro.io.Encoder
+import org.apache.avro.io.EncoderFactory
+import java.io.ByteArrayInputStream
+import java.io.ByteArrayOutputStream
+import java.io.InputStream
+
+
+sealed interface EnDecoder {
+    val name: String
+    var avro: Avro
+    fun encodeGenericRecordForComparison(value: GenericRecord, schema: Schema): ByteArray
+
+    fun <T> decode(
+        byteArray: ByteArray,
+        deserializer: DeserializationStrategy<T>,
+        readSchema: Schema,
+        writeSchema: Schema
+    ): T
+
+    fun <T> encode(value: T, serializer: SerializationStrategy<T>, schema: Schema): ByteArray
+}
+
+class AvroLibEnDecoder : EnDecoder {
+    override var avro: Avro = Avro.default
+    override val name: String = "AvroLibrary"
+    override fun <T> encode(value: T, serializer: SerializationStrategy<T>, schema: Schema): ByteArray {
+        val asRecord = avro.toRecord(serializer, schema, value)
+        return encodeGenericRecordForComparison(asRecord, schema)
+    }
+
+    override fun encodeGenericRecordForComparison(value: GenericRecord, schema: Schema): ByteArray {
+        val writer = GenericDatumWriter<GenericRecord>(schema)
+        val byteArrayOutputStream = ByteArrayOutputStream()
+        val encoder = avroLibEncoder(schema, byteArrayOutputStream)
+        writer.write(value, encoder)
+        encoder.flush()
+        return byteArrayOutputStream.toByteArray()
+    }
+
+    override fun <T> decode(
+        byteArray: ByteArray,
+        deserializer: DeserializationStrategy<T>,
+        readSchema: Schema,
+        writeSchema: Schema
+    ): T {
+        val input = ByteArrayInputStream(byteArray)
+        val reader = GenericDatumReader<GenericRecord>(writeSchema, readSchema)
+        val genericData = reader.read(null, avroLibDecoder(writeSchema, input))
+        return avro.fromRecord(deserializer,genericData)
+    }
+
+    fun avroLibEncoder(schema: Schema, outputStream: ByteArrayOutputStream): Encoder =
+        EncoderFactory.get().jsonEncoder(schema, outputStream)
+    
+    fun avroLibDecoder(schema: Schema, inputStream: InputStream) : Decoder = 
+        DecoderFactory.get().jsonDecoder(schema, inputStream)
+}
+
+inline fun <reified T> EnDecoder.testEncodeDecode(
+    value: T,
+    shouldMatch: RecordBuilderForTest,
+    serializer: KSerializer<T> = avro.serializersModule.serializer<T>(),
+    schema: Schema = avro.schema(serializer)
+) {
+    val encoded = testEncodeIsEqual(value, shouldMatch, serializer, schema)
+    testDecodeIsEqual(encoded, value,serializer, schema)
+}
```

Comment:
Here is what I'm talking about: taking the expected avro representation, encoding it with avro lib, and then use it to test our decoding part
```suggestion
inline fun <reified T> EnDecoder.testEncodeDecode(
    value: T,
    shouldMatch: RecordBuilderForTest,
    serializer: KSerializer<T> = avro.serializersModule.serializer<T>(),
    schema: Schema = avro.schema(serializer)
) {
    testEncodeIsEqual(value, shouldMatch, serializer, schema)
    testDecodeIsEqual(avroLibBinaryEncoding(shouldMatch.toAvro(schema)), value, serializer, schema)
}
```